test/gem_test.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+require "helper"
+
+class GemTest < ProtoBoeuf::Test
+  def test_can_be_required
+    Tempfile.create do |file|
+      file.write(<<~RUBY)
+        require "bundler/inline"
+
+        gemfile do
+          gem "protoboeuf", path: "#{__dir__}/..", require: true
+        end
+
+        ::ProtoBoeuf
+
+        # The following should auto/eagerload
+        ::ProtoBoeuf::CodeGen
+        ::ProtoBoeuf::Google::Api::FieldBehavior
+        ::ProtoBoeuf::Google::Protobuf::Any
+
+        exit 0
+      RUBY
+
+      file.flush
+
+      assert(system("ruby #{file.path}"), "Failed to require the gem")
```

Comment:
I think we should use `RbConfig.ruby` so that the subprocess uses the running ruby rather than whatever ruby is in the PATH (there will be less surprises if someone is testing out different ruby versions).
Also, we should pass separate args to avoid the shell.
```suggestion
      assert(system(RbConfig.ruby, file.path), "Failed to require the gem")
```

Reply:
I'm also tempted to to add `"-v",` to the command to give us a place that's showing us what warnings are generated just by loading our code.

Reply:
`-v` outputs a bunch of `RUBY_GC_*` stuff too:

```
RUBY_GC_MALLOC_LIMIT=96000000 (default value: 16777216)
RUBY_GC_MALLOC_LIMIT_MAX=96000000 (default value: 33554432)
RUBY_GC_OLDMALLOC_LIMIT=128000000000 (default value: 16777216)
RUBY_GC_OLDMALLOC_LIMIT_MAX=128000000000 (default value: 134217728)
/var/folders/y7/kzy9gfgj0jvcw52l4l6bqlcr0000gn/T/20250115-21263-ylk2tf:7: warning: possibly useless use of :: in void context
/var/folders/y7/kzy9gfgj0jvcw52l4l6bqlcr0000gn/T/20250115-21263-ylk2tf:10: warning: possibly useless use of :: in void context
/var/folders/y7/kzy9gfgj0jvcw52l4l6bqlcr0000gn/T/20250115-21263-ylk2tf:11: warning: possibly useless use of :: in void context
/var/folders/y7/kzy9gfgj0jvcw52l4l6bqlcr0000gn/T/20250115-21263-ylk2tf:12: warning: possibly useless use of :: in void context
```

I tried various `-W*` warning levels but couldn't get it to give me just the warnings but not the GC variable values.  I'll play with it a little more though.

Reply:
Oh, interesting.  Not sure where that's happening but I can work with that.  Thanks.